core/lib/basic_types/src/protocol_version.rs

```diff
@@ -125,7 +125,10 @@ impl ProtocolVersionId {
             ProtocolVersionId::Version25 => VmVersion::Vm1_5_0IncreasedBootloaderMemory,
             ProtocolVersionId::Version26 => VmVersion::VmGateway,
             ProtocolVersionId::Version27 => VmVersion::VmGateway,
-            ProtocolVersionId::Version28 => unreachable!("Version 28 is not yet supported"),
+
+            // Speculative VM version for the next protocol version to be used in the upgrade integration test etc.
+            // TODO: Must be changed when the protocol version is actually implemented!
+            ProtocolVersionId::Version28 => VmVersion::VmGateway,
```

Comment:
Is this the correct way to do things? I don't think the upgrade integration test would work otherwise, since it's supposed to run with the next protocol version after the upgrade.

Reply:
Shouldn't we use the new subversion (smth like VmEvmEmulator) here?

Reply:
Once we determine how to deal with the new VM version in `multivm`, I'd imagine that yes we'd have `VmVersion:: VmEvmEmulator` for protocol versions v27 and v28.